### PR TITLE
GH-663 Add private chat api

### PR DESCRIPTION
**Pull request by vLuckyyy**
image

Sorry, one of the inexperienced students merged a pull request that was not completed.

### CODE REVIEW

**Review by P1otrulla — APPROVED** [review]

File: eternalcore-api/src/main/java/com/eternalcode/core/feature/privatechat/PrivateChatEvent.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+public class PrivateChatEvent extends Event {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private final UUID sender;
+    private final UUID receiver;
+    private final String content;
```
Comment (Rollczi) [inline eternalcore-api/src/main/java/com/eternalcode/core/feature/privatechat/PrivateChatEvent.java:13]: Maybe mutable ?

**Review by Rollczi — CHANGES_REQUESTED** [review]
I think we can expose morę methods from Private message service (reply() and privateMessage())

File: eternalcore-api/src/main/java/com/eternalcode/core/feature/privatechat/PrivateChatEvent.java (hunk 1 above)
Comment (vLuckyyy) [reply eternalcore-api/src/main/java/com/eternalcode/core/feature/privatechat/PrivateChatEvent.java:13]: image

**Comment by vLuckyyy** [comment]
> I think we can expose morę methods from Private message service (reply() and privateMessage())

image

image

File: eternalcore-api-example/src/main/java/com/eternalcode/example/feature/privatechat/ApiPrivateChatListener.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    }
+
+    @EventHandler
+    public void overridePrivateChat(PrivateChatEvent event) {
+        String content = event.getContent();
+
+        if (content.contains("kurła!")) {
+            event.setContent("nie wolno przeklinać!");
+        }
```
Comment (imDMK) [inline eternalcore-api-example/src/main/java/com/eternalcode/example/feature/privatechat/ApiPrivateChatListener.java:42]: This is final release?

**Review by imDMK — APPROVED** [review]

File: eternalcore-api-example/src/main/java/com/eternalcode/example/feature/privatechat/ApiPrivateChatListener.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        Player senderPlayer = this.server.getPlayer(sender);
+        Player receiverPlayer = this.server.getPlayer(receiver);
+
+        for (Player player : this.server.getOnlinePlayers()) {
+            if (!player.hasPermission("privatechat.spy")) {
+                return;
+            }
+
+            player.sendMessage("Private Chat: " + senderPlayer.getName() + " -> " + receiverPlayer.getName() + ": " + content);
```
[the remark this answers is not in the document]
Comment (vLuckyyy) [reply eternalcore-api-example/src/main/java/com/eternalcode/example/feature/privatechat/ApiPrivateChatListener.java:32]: That is only an API example; no one besides the programmers manually inspecting the code on GitHub will be able to run it.

[the remark this answers is not in the document]
Comment (Rollczi) [reply eternalcore-api-example/src/main/java/com/eternalcode/example/feature/privatechat/ApiPrivateChatListener.java:32]: Real, resolved

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/privatechat/PrivateChatServiceImpl.java [hunk 4]
```diff
@@ -76,16 +87,29 @@ void reply(User sender, String message) {
...
         return this.socialSpy.contains(player);
     }
 
+    @Override
+    public void reply(Player sender, String message) {
+        this.reply(this.userManager.getOrCreate(sender.getUniqueId(), sender.getName()), message);
+    }
+
+    @Override
+    public void privateMessage(Player sender, Player target, String message) {
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/privatechat/PrivateChatServiceImpl.java:111]: To może trochę w przyszłości przeszkodzić w /msg przez console

**Review by Rollczi — APPROVED** [review]

**Pull request merged**